tests/src/e2e.rs

```diff
@@ -406,11 +411,246 @@ mod tests {
+            "intent",
+            "--node",
+            "http://127.0.0.1:39111",
+            "--data-path",
+            intent_b_path.as_ref(),
+            "--topic",
+            "asset_v1",
+        ]);
+        let mut session_send_intent_b =
+            spawn_command(send_intent_b, Some(20_000))
+                .map_err(|e| eyre!(format!("{}", e)))?;
+
+        // means it sent it correctly but not able to gossip it (which is
+        // correct since there is only 1 node)
+        session_send_intent_b
+            .exp_regex(".*Failed to publish_intent InsufficientPeers*")
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        session_gossip
+            .exp_regex(".*trying to match intent: *")
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        // // Send intent C
+        let mut send_intent_c = Command::cargo_bin("anomac")?;
+        send_intent_c.args(&[
+            "intent",
+            "--node",
+            "http://127.0.0.1:39111",
+            "--data-path",
+            intent_c_path.as_ref(),
+            "--topic",
+            "asset_v1",
+        ]);
+        let mut session_send_intent_c =
+            spawn_command(send_intent_c, Some(20_000))
+                .map_err(|e| eyre!(format!("{}", e)))?;
+
+        // means it sent it correctly but not able to gossip it (which is
+        // correct since there is only 1 node)
+        session_send_intent_c
+            .exp_regex(".*Failed to publish_intent InsufficientPeers*")
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        // check that the amount matched are correct
+        session_gossip
+            .exp_regex(".*amounts: [100000000, 70000000, 200000000]*")
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        // check that the transfers transactions are correct
+        session_gossip
+            .exp_regex(".*crafting transfer: Established: a1qq5qqqqqxv6yydz9xc6ry33589q5x33eggcnjs2xx9znydj9xuens3phxppnwvzpg4rrqdpswve4n9, Established: a1qq5qqqqqg4znssfsgcurjsfhgfpy2vjyxy6yg3z98pp5zvp5xgersvfjxvcnx3f4xycrzdfkak0xhx, 70000000*")
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        session_gossip
+            .exp_regex(".*crafting transfer: Established: a1qq5qqqqqxsuygd2x8pq5yw2ygdryxs6xgsmrsdzx8pryxv34gfrrssfjgccyg3zpxezrqd2y2s3g5s, Established: a1qq5qqqqqxv6yydz9xc6ry33589q5x33eggcnjs2xx9znydj9xuens3phxppnwvzpg4rrqdpswve4n9, 200000000*")
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        session_gossip
+            .exp_regex(".crafting transfer: Established: a1qq5qqqqqg4znssfsgcurjsfhgfpy2vjyxy6yg3z98pp5zvp5xgersvfjxvcnx3f4xycrzdfkak0xhx, Established: a1qq5qqqqqxsuygd2x8pq5yw2ygdryxs6xgsmrsdzx8pryxv34gfrrssfjgccyg3zpxezrqd2y2s3g5s, 100000000*")
+            .map_err(|e| eyre!(format!("{}", e)))?;
+
+        drop(session_gossip);
+        drop(session_ledger);
+        drop(session_send_intent_a);
+        drop(session_send_intent_b);
+        drop(session_send_intent_c);
```

Comment:
I think you can skip all the `drop`s as they will drop when they go out of scope and there's nothing happening after it

Reply:
idk if its related to this, but if I remove this series of `drop` the CI test e2e step doesn't stop and run infinity (https://ci.heliax.dev/anoma/anoma/703)